# Loading Libraries and Data
First, let us load the necessary libraries into the Jupyter Notebook, and load in the data we need from the players.csv and sessions.csv files.

In [16]:
library(tidyverse)
players_info <- read_csv("players.csv")
sessions_info <- read_csv("sessions.csv")
# head(players_info)
# head(sessions_info)
unique(players_info$gender)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] "Male"              "Female"            "Non-binary"       
[4] "Prefer not to say" "Agender"           "Two-Spirited"     
[7] "Other"

# 1) Data Description

In [26]:
summary(players_info)
# players_info |>
# count(experience)
# players_info |>
# count(gender)
players_info |>
filter(played_hours == 0)

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

experience subscribe
1  Veteran    FALSE    
2  Amateur     TRUE    
3  Regular     TRUE    
4  Amateur    FALSE    
5  Veteran     TRUE    
6  Pro         TRUE    
7  Veteran     TRUE    
8  Amateur    FALSE    
9  Beginner    TRUE    
10 Beginner    TRUE    
11 Regular     TRUE    
12 Veteran    FALSE    
13 Amateur     TRUE    
14 Amateur    FALSE    
15 Veteran     TRUE    
16 Amateur     TRUE    
17 Veteran     TRUE    
18 Beginner   FALSE    
19 Regular    FALSE    
20 Beginner   FALSE    
21 Regular     TRUE    
22 Regular     TRUE    
23 Veteran     TRUE    
24 Regular     TRUE    
25 Veteran     TRUE    
26 Veteran     TRUE    
27 Regular     TRUE    
28 Regular     TRUE    
29 Amateur     TRUE    
30 Veteran     TRUE    
⋮  ⋮          ⋮        
56 Amateur     TRUE    
57 Amateur    FALSE    
58 Beginner    TRUE    
59 Veteran     TRUE    
60 Regular     TRUE    
61 Veteran     TRUE    
62 Veteran     TRUE    
63 Veteran     TRUE    
64 Regular     TRUE    
65 Regular    FALSE    
66 Pro         TRUE    
67 Veteran     TRUE    
68 Amateur     TRUE    
69 Amateur    FALSE    
70 Amateur     TRUE    
71 Amateur     TRUE    
72 Beginner    TRUE    
73 Beginner    TRUE    
74 Veteran    FALSE    
75 Regular     TRUE    
76 Veteran     TRUE    
77 Regular    FALSE    
78 Veteran     TRUE    
79 Pro        FALSE    
80 Amateur     TRUE    
81 Beginner    TRUE    
82 Amateur    FALSE    
83 Amateur     TRUE    
84 Amateur     TRUE    
85 Amateur    FALSE    
   hashedEmail                                                     
1  b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28
2  f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977
3  8e594b8953193b26f498db95a508b03c6fe1c24bb5251d392c18a0da9a722807
4  1d2371d8a35c8831034b25bda8764539ab7db0f63938696917c447128a2540dd
5  bbe2d83de678f519c4b3daa7265e683b4fe2d814077f9094afd11d8f217039ec
6  4caa42e1b20511552434978171dcf7283fb6eb857eb871327e39bfc328002309
7  6f9acf8ea9956fe817895c78d10e1e25c11aba335a451e5b0322b073ee5008c4
8  2313a06afe47eacc28ff55adf6f072e7d12b0d12d7cbae29656f9029ec72bb53
9  9db37a1d381697f6c83e3b97b276e38bb9d9c811ba4596edb75d4dbfea6415b0
10 89c8e07d6a2aa102d9cd05086d51b2e41195c9c807dd976a06f4262ff36d0cca
11 50cec73a632e4be7b155c3df5165de28b2c5baf092abcb7f509f5645776a56c2
12 f8acd26a5e452b682b9f8b20108caef1ee2e745efe08e95806980bd2388a3fea
13 1683a3e0aed65119f83540274ff6f965fdf66890613a80f88572dfb1d9f1d2ad
14 df6a1201c9c638aff1df8aeaf459a5b0a00aa04824936ee0305f5c275352c647
15 d782933acd14c834e53dea816005a3583cb87710f7347a7a862cb4a43cc21692
16 6ef84a204d64edc3b3779127b13f298d0c70f96568486f785271f4c81ec0fe72
17 126ec350278422ba60ad575d230c19522fd568094b7c0fc17f8cc27fa4c810ab
18 fa7d496b2f74c51ec70395bd8397b49f97a3ce8d7ba7e08887d82ec0008137a7
19 453774822d52d6891593948224570a750c4334616ff8aa74cd61fad9d4855752
20 5e5c25a773be7a62638a163d773534e575a5ad5782104723e53cb676a30fa60b
21 e8df74a0605dedc2d6ee2d2651019f9006965f8b70223d5bcf9c26337eb6df05
22 dadce3d41f822c6303ac17f48b63699d7437cb7392acc63584628e5cd0346b5f
23 055d1a2002f6308a95b03b8d17af28f43e9f91ca8b1e1fc439bf4d08b22e7c0f
24 e2d5e72a6486ad8f83dd71163f44f88138612b9f40c17e9455910fb06bb45735
25 1edebbf13898dbe99da5ee743b1fd9cbfdc6d69f1180dd6261da662bacb1a7e5
26 a56b4b044e562fcf7765387d53c9e6543c42595fb4318627989942ab5979f9d6
27 e21a324ccf5c873bafe82e47d5137b36aa312ee4803eeb5cbc6cf945f9bb3f38
28 65c53ff7acdd69fd42b2a3b04e44870ea12b55b93a7ea1d6855c21a861b5a373
29 dbb20960cd4db4900dde7832e349dae46f7688583c8138b34785d9c7e509b5b9
30 2cbbfa45451a38b85663b7a9a3fffed12af1ba063ec03613d4c9e3873090cd05
⋮  ⋮                                                               
56 f6b7fa3cfaeb2f8d2cedf287cac000e219be75a0278039edc4c69eaf96250683
57 11bf6125c4264b3a8f3bffa57b33bd598e2ea1ecd6331ad3bc54aa88e22db035
58 7b5c323f9a9e21b6bcc2a8c431e96f9f2155c7b4318a68d3e7d9edd2018095a9
59 9a559f381d4c432cd528992b9c31f624b5b7074f642328c29d16fd2277bd1cc2
60 657e5c2da9dad8ec67b8f2875d98c290dc97573d86b77022d3b58ee913ed491c

Looking at the summary of the players.csv file, there are 196 observations and 7 variables, showing data for each player. Below is information for each of the variables.
- experience: character type, describing the experience level of each player, which can be either 'Beginner' (35 players), 'Regular' (36 players), 'Amateur' (63 players), 'Veteran' (48 players) or 'Pro' (14 players).
- subscribe: logical type, describing if the player is subscribed (True, 144 players) or not (False, 52 players).
- hashedEmail: character type, the email of the player, which is masked to protect their identity.
- played_hours: double type, the amount of hours played by a given player, ranging from 0 to 223.10 hours, with a mean of 5.8 hours.
- name: character type, the name of each player.
- gender: character type, the gender of each player, either 'Male' (124 players), 'Female' (37 players), 'Non-binary' (15 players), 'Prefer not to say' (11 players), 'Two-Spirited' (6 players), 'Agender' (2 players) or 'Other (1 player).
- age: double type, the age of the player, ranging from 9 to 58 years old, with a median age of 19 years old.

The biggest issue I saw is that there are 85 players who have played 0 hours, which may affect future data analyses.

In [24]:
# summary(sessions_info)
head(sessions_info)

hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

Looking at the summary of the players.csv file, there are 1535 observations and 5 variables, showing data about individual play sessions from each player. Below is information for each of the variables.
- hashedEmail: character type, the email of the player, which is masked to protect their identity.
- start_time: character type, the start time of each session
- end_time: character type, the end time of each session
- original_start_time: double type, the original start time of each session
- original_end_time: double type, the original end time of each session

The biggest issue I saw in this file was that both the original_start_time and original_end_time columns have numbers that are significantly large, and may be difficult to visualize.

# 2) Questions

What are

# 3) Exploratory Data Analysis and Visualization

Now that we have

In [10]:
print("hello")

[1] "hello"


# 4) Methods and Plan

The method

# 5) GitHub Repository Link

https://github.com/saurav-s25/DSCI-100-Project-Individual.git